src/hooks/useHasAppUpdated/useHasAppUpdated.ts

```diff
@@ -1,64 +1,49 @@
 import { useInterval } from '@chakra-ui/hooks'
 import axios from 'axios'
-import { useState } from 'react'
+import { useMemo, useState } from 'react'
+import { checkObjEquality } from 'lib/utils'
 
 import { getConfig } from '../../config'
 
 const APP_UPDATE_CHECK_INTERVAL = 1000 * 60
 
 export const useHasAppUpdated = () => {
   const [hasUpdated, setHasUpdated] = useState(false)
+  const [initialManifestMainJs, setManifestMainJs] = useState(null)
+  // asset-manifest tells us the latest minified built files
+  const url = '/asset-manifest.json'
+  const storeMainManifestJs = async () => {
+    // dummy query param bypasses browser cache
+    const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
+    setManifestMainJs(data)
+  }
+  useMemo(storeMainManifestJs, [])
   useInterval(async () => {
     // we don't care about updates locally obv
     if (getConfig().isDevelopment) return
-    // this will break if we ever eject from create react app
-    const scriptIdentifier = '/static/js/main.'
-    // asset-manifest tells us the latest minified built files
-    const url = '/asset-manifest.json'
+
     let manifestMainJs
     try {
-      // dummy query param bypasses browser cache
       const { data } = await axios.get(`${url}?${new Date().valueOf()}`)
-      manifestMainJs = data?.files?.['main.js']
+      manifestMainJs = data
     } catch (e) {
       console.error(`useHasAppUpdated: error fetching asset-manifest.json`, e)
     }
     if (!manifestMainJs) {
       console.error(`useHasAppUpdated: can't find main.js in asset-manifest.json`)
       return
     }
-    if (!manifestMainJs.includes(scriptIdentifier)) {
-      console.error(
-        `useHasAppUpdated: manifest main.js doesn't start with identifier ${scriptIdentifier}`,
-        manifestMainJs
+    //deep equality check
+    let areEquals = checkObjEquality(initialManifestMainJs, manifestMainJs)
+    if (!areEquals) {
```

Comment:
can we change this to be named `isSameAssetManifest` please?

Reply:
yes! no problem :)

Reply:
it's done!